pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -245,9 +305,247 @@ func (r *ReconcileNodeMaintenance) stopNodeMaintenance(nodeName string) error {
+		Name:       nodeName,
+		UID:        node.ObjectMeta.UID,
+	}
+	if lease == nil {
+		leaseTransitions := int32(1)
+		microTimeNow := metav1.NewMicroTime(time.Now())
+
+		var refDuration *int32 = nil
+		if tmpDurationInSeconds != 0 {
+			refDuration = &tmpDurationInSeconds
+		}
+
+		lease = &coordv1beta1.Lease{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            nodeName,
+				Namespace:       LeaseNamespace,
+				OwnerReferences: []metav1.OwnerReference{*owner},
+			},
+			Spec: coordv1beta1.LeaseSpec{
+				HolderIdentity:       &holderIdentity,
+				LeaseDurationSeconds: refDuration,
+				AcquireTime:          &microTimeNow,
+				RenewTime:            &microTimeNow,
+				LeaseTransitions:     &leaseTransitions,
+			},
+		}
+
+		if err := r.client.Create(context.TODO(), lease); err != nil {
+			reqLogger.Errorf("Failed to create lease. node %s error: %v", nodeName, err)
+			return lease, err
+		}
+		reqLogger.Debugf("lease created. duration: %d sec", tmpDurationInSeconds)
+	} else {
+		lease.ObjectMeta.Name = nodeName
+		lease.ObjectMeta.OwnerReferences = []metav1.OwnerReference{*owner}
+		lease.Spec.HolderIdentity = &holderIdentity
+		timeNow := metav1.MicroTime{Time: time.Now()}
+		lease.Spec.AcquireTime = &timeNow
+		lease.Spec.RenewTime = &timeNow
+
+		if transitions == TransitionInc && lease.Spec.LeaseTransitions != nil {
+			*lease.Spec.LeaseTransitions += int32(1)
+		} else if transitions != TransitionNoChange {
+			leaseTransitions := int32(1)
+			lease.Spec.LeaseTransitions = &leaseTransitions
+		}
+
+		var refDuration *int32 = nil
+		if tmpDurationInSeconds != 0 {
+			refDuration = &tmpDurationInSeconds
+		}
+		lease.Spec.LeaseDurationSeconds = refDuration
+
+		if err := r.client.Update(context.TODO(), lease); err != nil {
+			reqLogger.Errorf("Failed to update the lease. node %s error: %v", nodeName, err)
+			return lease, err
+		}
+		reqLogger.Debugf("lease updated: duration: %d sec", tmpDurationInSeconds)
+	}
+
+	return lease, nil
+}
+
+// returns true if lease duration differs from node annotation value
+func (r *ReconcileNodeMaintenance) isLeaseDurationChanged(lease *coordv1beta1.Lease, durationValue int) bool {
+	if lease.Spec.LeaseDurationSeconds != nil {
+		return *lease.Spec.LeaseDurationSeconds != int32(durationValue)+LeasePaddingSeconds
+	}
+	return false
+}
+
+func isLeaseDurationZero(lease *coordv1beta1.Lease) bool {
+
+	return lease.Spec.LeaseDurationSeconds == nil || *lease.Spec.LeaseDurationSeconds == 0
+}
+
+func leasePeriodEnd(lease *coordv1beta1.Lease) time.Time {
+
+	if lease.Spec.AcquireTime != nil || lease.Spec.RenewTime != nil {
+
+		var leasePeriodEnd time.Time
+
+		if lease.Spec.RenewTime != nil {
+			leasePeriodEnd = (*lease.Spec.RenewTime).Time
+		} else {
+			leasePeriodEnd = (*lease.Spec.AcquireTime).Time
+		}
+
+		if lease.Spec.LeaseDurationSeconds != nil {
+			leasePeriodEnd.Add(time.Duration(*lease.Spec.LeaseDurationSeconds) * time.Second)
+		}
+		return leasePeriodEnd
+	}
+	return time.Now()
```

Comment:
I don't think this is a good default value to return

Reply:
removed the unused function.